lib/entities/quote/RelayQuote.ts

```diff
@@ -0,0 +1,304 @@
+import { RelayOrder, RelayOrderBuilder, RelayOrderInfoJSON } from '@uniswap/uniswapx-sdk';
+import { SwapRouter, UniswapTrade, UNIVERSAL_ROUTER_ADDRESS } from '@uniswap/universal-router-sdk';
+import { BigNumber, ethers } from 'ethers';
+
+import { PermitBatchTransferFromData } from '@uniswap/permit2-sdk';
+import { Percent } from '@uniswap/sdk-core';
+import { IQuote } from '.';
+import { DEFAULT_START_TIME_BUFFER_SECS, RELAY_BASE_GAS, RoutingType } from '../../constants';
+import { generateRandomNonce } from '../../util/nonce';
+import { currentTimestampInMs, timestampInMstoSeconds } from '../../util/time';
+import { RelayRequest } from '../request/RelayRequest';
+import { ClassicQuote, ClassicQuoteDataJSON } from './ClassicQuote';
+import { LogJSON } from './index';
+
+// Data returned by the API
+export type RelayQuoteDataJSON = {
```

Comment:
Given there's a lot of overlap in the different QuoteData types, should we create a common type that these can extend?

Reply:
Ah yes good call out, I've made a shared type for all non classic quotes (X, Xv2, relay) and moved some of the shared fields into the type.